src/main/cpp/util/file_windows.cc

```diff
@@ -548,9 +552,9 @@ bool CanAccessDirectory(const Path& path) {
       /* dwFlagsAndAttributes */ FILE_ATTRIBUTE_NORMAL,
       /* hTemplateFile */ NULL);
   DWORD err = GetLastError();
-  if (handle == INVALID_HANDLE_VALUE && err != ERROR_ALREADY_EXISTS) {
```

Comment:
We should respect any error here, `handle == INVALID_HANDLE_VALUE && err==ERROR_ALREADY_EXISTS` is incorrect invariant for `OPEN_ALWAYS`.

Reply:
Why incorrect? CreateFileW's docs says: "If the specified file exists, the function succeeds and the last-error code is set to ERROR_ALREADY_EXISTS (183)."

Reply:
Yes, but if function succeeds handle != INVALID_HANDLE_VALUE

Reply:
Ah, of course.